src/Http/Requests/Bulk/DestroyPosts.php

```diff
@@ -35,7 +35,7 @@ public function fulfill()
     {
         $posts = $this->posts();
 
-        if (config('forum.general.soft_deletes') && isset($this->validated()['permadelete']) && $this->validated()['permadelete'] && method_exists(Post::class, 'forceDelete'))
+        if (config('forum.general.soft_deletes') && $this->isPermaDeleteRequested() && method_exists(Post::class, 'forceDelete'))
         {
             $post->forceDelete();
```

Comment:
Looks like it's supposed to be in a loop - I'm also not sure this would work well with the code that follows (updating associated threads and categories). Will look at this separately.